chain/client/src/chunk_validation.rs

```diff
@@ -261,18 +267,76 @@ fn pre_validate_chunk_state_witness(
         )));
     }
 
+    let block = Chain::get_apply_chunk_block_context(
+        epoch_manager,
+        last_chunk_block.header(),
+        &store.get_previous_header(last_chunk_block.header())?,
+        true,
+    )?;
+    let epoch_id = epoch_manager.get_epoch_id(&block.block_hash)?;
+    let chunk_header = last_chunk_block.chunks().get(shard_id as usize).unwrap().clone();
```

Comment:
please rename `chunk_header` => `prev_chunk_header` or `last_chunk_header`
but also this header should not be used for transactions validation, so maybe just inline that back to `NewChunkData `